webgpu.c

```diff
@@ -5,1021 +5,897 @@
 
 #include "imports.h"
 
+typedef struct WGPUAdapterImpl {
+    wasi_webgpu_webgpu_own_gpu_adapter_t adapter;
+} WGPUAdapterImpl;
+// typedef struct WGPUBindGroupImpl {
+// } WGPUBindGroupImpl;
+// typedef struct WGPUBindGroupLayoutImpl {
+// } WGPUBindGroupLayoutImpl;
+// typedef struct WGPUBufferImpl {
+// } WGPUBufferImpl;
+// typedef struct WGPUCommandBufferImpl {
+// } WGPUCommandBufferImpl;
+// typedef struct WGPUCommandEncoderImpl {
+// } WGPUCommandEncoderImpl;
+// typedef struct WGPUComputePassEncoderImpl {
+// } WGPUComputePassEncoderImpl;
+// typedef struct WGPUComputePipelineImpl {
+// } WGPUComputePipelineImpl;
+typedef struct WGPUDeviceImpl {
+    wasi_webgpu_webgpu_own_gpu_device_t device;
+} WGPUDeviceImpl;
+typedef struct WGPUInstanceImpl {
+    wasi_webgpu_webgpu_own_gpu_t gpu;
+} WGPUInstanceImpl;
+// typedef struct WGPUPipelineLayoutImpl {
+// } WGPUPipelineLayoutImpl;
+// typedef struct WGPUQuerySetImpl {
+// } WGPUQuerySetImpl;
+// typedef struct WGPUQueueImpl {
+// } WGPUQueueImpl;
+// typedef struct WGPURenderBundleImpl {
+// } WGPURenderBundleImpl;
+// typedef struct WGPURenderBundleEncoderImpl {
+// } WGPURenderBundleEncoderImpl;
+// typedef struct WGPURenderPassEncoderImpl {
+// } WGPURenderPassEncoderImpl;
+// typedef struct WGPURenderPipelineImpl {
+// } WGPURenderPipelineImpl;
+// typedef struct WGPUSamplerImpl {
+// } WGPUSamplerImpl;
+// typedef struct WGPUShaderModuleImpl {
+// } WGPUShaderModuleImpl;
+// typedef struct WGPUSurfaceImpl {
+// } WGPUSurfaceImpl;
+// typedef struct WGPUTextureImpl {
+// } WGPUTextureImpl;
+// typedef struct WGPUTextureViewImpl {
+// } WGPUTextureViewImpl;
 
 WGPUInstance wgpuCreateInstance(WGPUInstanceDescriptor const* descriptor)
 {
-    exit(1);
+    WGPUInstanceImpl* instance = malloc(sizeof(WGPUInstanceImpl));
```

Comment:
Make sure there are associated free(s) / resource cleanup methods in future features, and you could consider using RAII here if you don't need the implementation to be in pure C.

Reply:
@seanisom like [`wgpuInstanceRelease`](https://github.com/webgpu-native/webgpu-headers/blob/5bca022bca04745cbfdbc605e9db1d7ae2357d7e/webgpu.h#L3086)?

Reply:
@seanisom I've added reference counting